File: easy/13/roman_to_integer.py

class Solution:
    def romanToInt(self, s: str) -> int:
        
        roman_d = { 'I': 1, 'V': 5, 'X': 10, 'L': 50, 'C': 100, 'D': 500, 'M': 1000 }
        p = roman_d[s[0]]
        total = p

        for i in range(1,len(s)):
            c = roman_d[s[i]]
            total += c - (2 * p) if p < c else c
            p = c
        
        return total

File: easy/13/roman_to_integer.cpp

#include <map>

typedef std::map<char,int> dict;

class Solution {
public:
    int romanToInt(string s) {
        int p, c, total;
        dict roman_d;

        roman_d['I'] = 1;
        roman_d['V'] = 5;
        roman_d['X'] = 10;
        roman_d['L'] = 50;
        roman_d['C'] = 100;
        roman_d['D'] = 500;
        roman_d['M'] = 1000;
        
        p = roman_d[s[0]];
        total = p;
        
        for (int i = 1; i < s.size(); i++) {
            c = roman_d[s[i]];
            total += (p < c) ? c - (2 * p) : c; 
            p = c;
        }
        
        return total;
    }
};
